examples/huggingfacebert/server.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Example script for hosting a BERT ONNX model as a Flask server
+using the DeepSparse Engine as the inference backend
+
+##########
+Command help:
+usage: server.py [-h] [-b BATCH_SIZE] [-c NUM_CORES] [-a ADDRESS] [-p PORT]
+                 [-q]
+                 onnx_filepath
+
+Host a BERT ONNX model as a server, using the DeepSparse Engine and Flask
+
+positional arguments:
+  onnx_filepath         The full filepath of the ONNX model file or SparseZoo
+                        stub to the model
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -b BATCH_SIZE, --batch-size BATCH_SIZE
+                        The batch size to run the analysis for
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the analysis on,
+                        defaults to all physical cores available on the system
+  -a ADDRESS, --address ADDRESS
+                        The IP address of the hosted model
+  -p PORT, --port PORT  The port that the model is hosted on
+##########
+Example command for running:
+python server.py \
+    ~/models/bert-base-uncased.onnx
+"""
+import argparse
+import json
+import sys
+import time
+
+import flask
+from flask import Flask, jsonify, make_response
+from flask_cors import CORS
+
+from deepsparse.transformers import pipeline
+
+
+app = Flask(__name__)
+CORS(app)
+
+
+def run_server(predictor, host="0.0.0.0", port="5543", info=None):
+    """
+    Method to create routes and serve predictor on the specified port
+    """
+    assert callable(predictor), "predictor should be callable"
+
+    @app.route("/predict", methods=["POST"])
+    def predict():
+        """
+        Expects data in json format with question and context
+
+        '{"question":"what's my name?","context":"my name is snorlax"}'
+
+        :returns: A json with results from predictor and the inputs
+        """
+        data = json.loads(flask.request.get_data())
```

Comment:
let's add quick input validation that `"question"` and `"context"` are present and both are either a string or list of strings